Source/Project64/N64 System/Mips/Memory Labels Class.cpp

```diff
@@ -23,33 +23,25 @@ DWORD CMemoryLabel::AsciiToHex (char * HexValue)
 	for (Count = 0; Count < Finish; Count++)
 	{
 		Value = (Value << 4);
-		switch ( HexValue[Count] )
+		Current = HexValue[Count];
+		if(Current >= '0' && Current <= '9')
 		{
-		case '0': break;
-		case '1': Value += 1; break;
-		case '2': Value += 2; break;
-		case '3': Value += 3; break;
-		case '4': Value += 4; break;
-		case '5': Value += 5; break;
-		case '6': Value += 6; break;
-		case '7': Value += 7; break;
-		case '8': Value += 8; break;
-		case '9': Value += 9; break;
-		case 'A': Value += 10; break;
-		case 'a': Value += 10; break;
-		case 'B': Value += 11; break;
-		case 'b': Value += 11; break;
-		case 'C': Value += 12; break;
-		case 'c': Value += 12; break;
-		case 'D': Value += 13; break;
-		case 'd': Value += 13; break;
-		case 'E': Value += 14; break;
-		case 'e': Value += 14; break;
-		case 'F': Value += 15; break;
-		case 'f': Value += 15; break;
-		default: 
-			Value = (Value >> 4);
-			Count = Finish;
+			Value += Current - '0';
+		}
+		else
+		{
+			if(Current < 'A')
+				Current += 'A' - 'a';

```

Comment:
There was one question I had been wondering here.

In ASCII, `'A' - 'a'` evaluates to a negative number, because `('A' < 'a')`.
So I think instead of converting uppercase to lowercase, this converts lowercase to non-printables.

But better algorithm in principle overall anyway so good job.  I'll let anyone else decide if it's an issue.